binary_search/binary_search.java

```diff
@@ -0,0 +1,46 @@
+/**
+ *
+ * @author jsroyal
+ */
+public class BinarySearch {
+
+    public static void main(String[] args) {
+        int arr[] = {1, 5, 35, 112, 258, 324},
+                search_arr[] = {1, 35, 112, 324, 67},
```

Comment:
According to `Java` conventions, local variable names do not contain underscores (`_`).